lib/recurly/subscription/add_ons.rb

```diff
@@ -27,7 +27,7 @@ def << add_on
 
         exist = @add_ons.find { |a| a.add_on_code == add_on.add_on_code }
         if exist
-          exist.quantity ||= 1 and exist.quantity += 1
+          exist.quantity ||= 1 and exist.quantity += add_on.quantity || 1
```

Comment:
Hi,

I think the oneliner works as expected but maybe I haven't tested all the cases. :)
I'll refine this to use the 2 lines versions which is correct for sure.